packages/hoppscotch-cli/src/utils/mutators.ts

```diff
@@ -0,0 +1,93 @@
+import fs from "fs/promises";
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import * as A from "fp-ts/Array";
+import * as RA from "fp-ts/ReadonlyArray";
+import { pipe } from "fp-ts/function";
+import { RequestStack } from "../interfaces";
+import { FormDataEntry, error, HoppCLIError } from "../types";
+import { isRESTCollection, requestsParser } from ".";
+import { HoppCollection, HoppRESTRequest } from "@hoppscotch/data";
+
+/**
+ * Parses array of FormDataEntry to FormData.
+ * @param values Array of FormDataEntry.
+ * @returns FormData with key-value pair from FormDataEntry.
+ */
+export const toFormData = (values: FormDataEntry[]) => {
+  const formData = new FormData();
+
+  values.forEach(({ key, value }) => formData.append(key, value));
+
+  return formData;
+};
+
+/**
+ * Parses provided error message to maintain hopp-error messages.
+ * @param e Custom error data.
+ * @returns Parsed error message without extra spaces.
+ */
+export const parseErrorMessage = (e: any) => {
+  let msg: string;
+  if (e instanceof Error) {
+    const x = e as NodeJS.ErrnoException;
```

Comment:
I don't understand this cast, what if the error is not an `ErrnoException`, why can't we instead directly do instanceof `e` to `ErrnoException`.

Reply:
Actually, `ErrnoException` is just an interface within `NodeJS` namespace, but `ErrnoException` extends `Error` interface.
And the `Error` here is the decalaration of `ErrorConstructor` (which initiates Error related props),
And thats why this way to use `ErrnoException` has it has more properties which are usefull to create error message.

Reply:
aaah gotcha. that makes sense, I would suggest writing a utility type guard for it though, this entire function looks kinda funky.